libs/components/router/src/lib/modules/href/href.directive.ts

```diff
@@ -44,6 +45,21 @@ export class SkyHrefDirective {
     return this.#_skyHref;
   }
 
+  /**
+   * A collection of query URL parameters.
+   */
+  @Input()
+  public set queryParams(value: SkyHrefQueryParams | undefined) {
+    if (value !== this.#_queryParams) {
+      this.#_queryParams = value;
+      this.#applyChanges(this.#getChanges());
```

Comment:
I added a setter to this input so that we could update the href when query parameters change.